cs/commandstation/ProgrammingTrackFrontend.hxx

```diff
@@ -295,6 +298,7 @@ class ProgrammingTrackFrontend
 
   Action enter_service_mode_done() {
     auto b = get_buffer_deleter(full_allocation_result(backend_));
+    if (b->data()->hasShortCircuit_) return call_immediately(STATE(pgm_short));
```

Comment:
This is your code in your repo, you do what you want, but I would have probably structured it with a ternary operator. I would have found that to be more readable:

`return (b->data()->hasShortCircuit_) ? call_immediately(STATE(pgm_short) : call immediately(STATE(act_service_mode);`

Same below...

Reply:
I find this is easier to read. Early exit upon error conditions in a way that's consistent across the entire class is a good pattern. It also shows opportunities where with a better infrastructure we could refactor the code.